java8-pt/JavaParser.g4

```diff
@@ -516,11 +516,17 @@ primary
     ;
 
 methodReference
-    : (qualifiedName | typeType | (qualifiedName '.')? SUPER | THIS) '::' typeArguments? IDENTIFIER
+    : (typeType | qualifiedName ('.' (SUPER | THIS | methodReferenceMethodCall))?
+      | (SUPER | THIS | methodReferenceMethodCall))
+      '::' typeArguments? IDENTIFIER
     | classType '::' typeArguments? NEW
     | typeType '::' NEW
     ;
 
+methodReferenceMethodCall
+    : IDENTIFIER LPAREN expressionList? RPAREN (DOT IDENTIFIER LPAREN expressionList? RPAREN)*
+    ;
+
```

Comment:
I'm not an expert in Java. But in my opinion it can be rewritten more clear:

```ANTLR
methodReference
    : (typeType | qualifiedName ('.' accessorOrMethodRefCall)? | accessorOrMethodRefCall) '::' typeArguments? IDENTIFIER
    | classType '::' typeArguments? NEW
    | typeType '::' NEW
    ;

accessorOrMethodRefCall
    : SUPER
    | THIS
    | IDENTIFIER LPAREN expressionList? RPAREN (DOT IDENTIFIER LPAREN expressionList? RPAREN)*
    ;
```